src/main/java/seedu/linkedout/logic/parser/SortCommandParser.java

```diff
@@ -0,0 +1,56 @@
+package seedu.linkedout.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.linkedout.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.linkedout.logic.parser.CliSyntax.PREFIX_FIELD;
+import static seedu.linkedout.logic.parser.CliSyntax.PREFIX_ORDER;
+
+import java.util.stream.Stream;
+
+import seedu.linkedout.logic.commands.SortCommand;
+import seedu.linkedout.logic.parser.exceptions.ParseException;
+import seedu.linkedout.model.applicant.Order;
+import seedu.linkedout.model.applicant.SortComparator;
+
+/**
+ * Parses input arguments and creates a new SortCommand object
+ */
+public class SortCommandParser implements Parser<SortCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SortCommand
+     * and returns an SortCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SortCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_FIELD, PREFIX_ORDER);
+
+        // either not present will throw ParseException
+        if (!arePrefixesPresent(argMultimap, PREFIX_ORDER, PREFIX_FIELD)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        // get().isBlank to see if String is present anot
+        // pass upper case to SortCommand
+        if (!argMultimap.getValue(PREFIX_ORDER).get().isBlank()
```

Comment:
is it better to declare the booleans before using it because it seems a bit long?
```suggestion
        boolean hasOrder = !argMultimap.getValue(PREFIX_ORDER).get().isBlank()
 
```